### ISSUE TITLE
Make `Lint/UselessMethodDefinition` alias-aware

### ISSUE THREAD

**Issue by vaporyhumo** [rubocop/rubocop#9198]
## Is your feature request related to a problem? Please describe.

When aliasing methods, I prefer to have the original method being referenced explicitly defined in the file, sometimes just calling `super`. This of course triggers the `Lint/UselessMethodDefinition` cop, which seems kinda strict for this case.

## Describe the solution you'd like

I'd like that `Lint/UselessMethodDefinition` checks for an alias before reporting the definition as useless, as aliasing an inherited method implicitly might be too obscure sometimes.

## Describe alternatives you've considered

I think this could be an optional feature, as an option for said cop. (`IgnoreAliasedMethods : true|false`?)

## Additional context

This raises `Lint/UselessMethodDefinition`, I wish it didn't.

```
def index
  super
end
alias show_all index
```

**Comment by bbatsov** [comment]
@fatkodima Please look into this when you can.

**Comment by marcandre** [comment]
> I prefer to have the original method being referenced explicitly defined in the file

May I ask why? I'm not sure I see how that adds any information.

**Comment by vaporyhumo** [comment]
> > I prefer to have the original method being referenced explicitly defined in the file
> 
> May I ask why? I'm not sure I see how that adds any information.

I don't think it **adds** information per se, just make it a little bit more at hand, most aliases I've seen are directly defined below the method itself. I consider this a nice-to-have only.

**Comment by marcandre** [comment]
Thanks.
If that code was in a PR for `rubocop`, I would ask for a good justification to add an alias *on a subclass* (maybe because I tend not to create aliases at all) of if `def show_all; index; end` is not better than an alias here (in case a subclass refines `index`). I would definitely ask you to remove this method definition as I find it more confusing than helpful (I have to parse these lines of code that do nothing, I'll re-read them trying to see if there's anything subtle I'm missing).

That being said, I'm not reviewing that code 😆 and the cost of a false-negative in this case is so low that so I don't really object to this 😅

**Comment by vaporyhumo** [comment]
Yeah I wouldn't do it in a library either. Only use case so far has been rails pundit policies, where using alias and inheritance has been really semantically useful for us. Since this is not by any means urgent for us, I'll put up PR when/if I have some time. Thanks for answering! You can close the issue if you want to declutter; I con open it again if I decide to go on with the PR.
